### PR TITLE
Config adding mechanism #55

### PR DESCRIPTION
**Pull request by fedyafedko**

### CODE REVIEW

File: StudyHub/StudyHub.Common/Configs/UserInvitationConfig.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿namespace StudyHub.Common.Configs;
```
Comment (Podgore) [inline StudyHub/StudyHub.Common/Configs/UserInvitationConfig.cs:1]: empty line

File: StudyHub/StudyHub.Common/Configs/ConfigBase.cs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿namespace StudyHub.Common.Configs;
```
Comment (Podgore) [inline StudyHub/StudyHub.Common/Configs/ConfigBase.cs:1]: empty line

**Review by Podgore — CHANGES_REQUESTED** [review]

**Review by Podgore — DISMISSED** [review]

File: StudyHub/StudyHub/Program.cs [hunk 3]
```diff
@@ -23,19 +22,18 @@
...
 using StudyHub.Utility;
 
 var builder = WebApplication.CreateBuilder(args);
 
-builder.Services.Configure<JwtSettings>(builder.Configuration.GetSection("JwtSettings"));
-builder.Services.Configure<GoogleAuthConfig>(builder.Configuration.GetSection("GoogleAuth"));
-builder.Services.Configure<EmailSettings>(builder.Configuration.GetSection("EmailSettings"));
-builder.Services.Configure<HangfireConfig>(builder.Configuration.GetSection("HangfireConfig"));
-builder.Services.Configure<UserInvitationConfig>(builder.Configuration.GetSection("UserInvitationConfig"));
+builder.Services.AddConfig<JwtConfig>(builder.Configuration);
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Program.cs:32]: I think we can avoid passing the same builder.Configuration for every AddConfig call

File: StudyHub/StudyHub/Extensions/ConfigExtension.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using StudyHub.Common.Configs;
+
+namespace StudyHub.Extensions;
+
+public static class ConfigExtension
+{
+    public static IServiceCollection AddConfig<T>(this IServiceCollection services, IConfiguration configuration)
+        where T : ConfigBase
+    {
+        return services.Configure<T>(configuration.GetSection(typeof(T).Name));
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Extensions/ConfigExtension.cs:10]: this adds config to DI as IOptions\<T\>, maybe we can directly add Config instance

**Review by aquaprogit — APPROVED** [review]

**Pull request merged**